orderbook/src/orderbook.rs

```diff
@@ -1,15 +1,18 @@
-use model::{Order, OrderCreation, OrderMetaData, OrderUid};
-use primitive_types::{H160, H256};
+use model::{Order, OrderCreation, OrderMetaData};
 use tokio::sync::RwLock;
 
 #[derive(Debug, Eq, PartialEq)]
 pub enum AddOrderError {
-    AlreadyExists,
+    DuplicatedOrder,
     InvalidSignature,
     #[allow(dead_code)]
-    PastNonce,
+    Forbidden,
+    #[allow(dead_code)]
+    MissingOrderData,
```

Comment:
Would be nice if this could take an argument with the (first) field name that is missing.

Reply:
Yeah, I agree. I tried it for over 1 hour, but my rust knowledge is too limited. Hence, I am giving up and focusing on unblocking the front-end.